From c2002948d9d6f9b82180f786bcfa601b1cf4fa87 Mon Sep 17 00:00:00 2001
From: Yipeng Zou <zouyipeng@huawei.com>
Date: Tue, 18 Apr 2023 15:23:00 +0000
Subject: [PATCH 0095/1646] [Huawei] Revert "[Huawei] genirq: Remove
 irqd_irq_disabled in __irq_move_irq"

Offering: HULK
hulk inclusion
category: bugfix
bugzilla: https://gitee.com/openeuler/kernel/issues/I6WPFT

--------------------------------

This reverts commit 1145c4bf06184bfdf0c5860f7c92584e3f9ed878.

Signed-off-by: Yipeng Zou <zouyipeng@huawei.com>
---
 kernel/irq/migration.c | 2 --
 1 file changed, 2 deletions(-)

diff --git a/kernel/irq/migration.c b/kernel/irq/migration.c
index 980ad6204cad..def48589ea48 100644
--- a/kernel/irq/migration.c
+++ b/kernel/irq/migration.c
@@ -102,10 +102,8 @@ void __irq_move_irq(struct irq_data *idata)
 	 */
 	idata = irq_desc_get_irq_data(irq_data_to_desc(idata));
 
-#ifndef CONFIG_ARM64
 	if (unlikely(irqd_irq_disabled(idata)))
 		return;
-#endif
 
 	/*
 	 * Be careful vs. already masked interrupts. If this is a
-- 
2.17.1

